samples/buildstrategy/buildah/buildstrategy_buildah_cr.yaml

```
@@ -17,9 +17,34 @@ spec:
         - |
           set -euo pipefail
 
+          echo "Creating registries config file..."
+
+          formate(){
+              array=(`echo $1 | tr ',' ' '`)
+              str=""
+              for m in "${array[@]}"; do
+                  str=$str"'${m}', "
+              done
+
+              echo ${str%??}
+          }
+
+          path="/tmp/registries.conf"
+          registry_search="$(formate "$(params.registry-search)")"
+          registry_insecure="$(formate "$(params.registry-insecure)")"
+          registry_block="$(formate "$(params.registry-block)")"
+
+          echo -e "[registries.search]" >> "$path"
+          echo -e "registries = [${registry_search}]\n" >> "$path"
+          echo -e "[registries.insecure]" >> "$path"
+          echo -e "registries = [${registry_insecure}]\n" >> "$path"
+          echo -e "[registries.block]" >> "$path"
+          echo -e "registries = [${registry_block}]\n" >> "$path"
+
           # Building the image
           echo '[INFO] Building image $(params.shp-output-image)'
           buildah bud \
+            --registries-conf='/tmp/registries.conf' \
```

Nit: When creating files on the spot, I like to use here-doc since this makes is more readable and easier to control the full look of the file. Also, the variable name `path` could lead to confusion with the `PATH` variable name.
```suggestion
          cat <<EOF >/tmp/registries.conf
          [registries.search]
          registries = [$(formate "$(params.registry-search)")]
          
          [registries.insecure]
          registries = [$(formate "$(params.registry-insecure)")]
          
          [registries.block]
          registries = [$(formate "$(params.registry-block)")]
          
          EOF

          # Building the image
          echo '[INFO] Building image $(params.shp-output-image)'
          buildah bud \
            --registries-conf='/tmp/registries.conf' \
```

---

Done